pkg/restapi/webhook/webhook_test.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webhook
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type clientData struct {
+	clientHost                     string
+	clientHandlerPattern           string
+	subscriberReceivedNotification chan struct{}
+}
+
+func TestWebhookDispatcherOneWebhook(t *testing.T) {
+	testClientData := clientData{
+		clientHost:                     randomURL(),
+		clientHandlerPattern:           "/webhookListen1",
+		subscriberReceivedNotification: make(chan struct{}),
+	}
+
+	go func() {
+		err := listenAndStopAfterReceivingNotification(testClientData.clientHost, testClientData.clientHandlerPattern)
+		if err != nil {
+			require.FailNow(t, err.Error())
+		} else {
+			close(testClientData.subscriberReceivedNotification)
+		}
+	}()
+
+	if err := listenFor(testClientData.clientHost); err != nil {
+		t.Fatal(err)
+	}
+
+	StartWebhookDispatcher([]string{fmt.Sprintf("http://%s%s", testClientData.clientHost,
+		testClientData.clientHandlerPattern)})
+
+	select {
+	case <-testClientData.subscriberReceivedNotification:
+		// Pass
+	case <-time.After(5 * time.Second):
+		require.FailNow(t, "samplewebhookclient did not receive a notification")
+	}
+}
+
+func TestWebhookDispatcherMultipleWebhooks(t *testing.T) {
+	clientData1 := clientData{
+		clientHost:                     randomURL(),
+		clientHandlerPattern:           "/webhookListen2",
+		subscriberReceivedNotification: make(chan struct{}),
+	}
+	clientData2 := clientData{
+		clientHost:                     randomURL(),
+		clientHandlerPattern:           "/webhookListen3",
+		subscriberReceivedNotification: make(chan struct{}),
+	}
+
+	allTestClientData := []clientData{clientData1, clientData2}
+
+	for _, testClientData := range allTestClientData {
+		go func() {
+			err := runClient(testClientData)
+			if err != nil {
+				require.FailNow(t, err.Error())
+			}
+		}()
+
+		if err := listenFor(testClientData.clientHost); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	StartWebhookDispatcher([]string{fmt.Sprintf("http://%s%s", allTestClientData[0].clientHost,
+		allTestClientData[0].clientHandlerPattern),
+		fmt.Sprintf("http://%s%s", allTestClientData[1].clientHost, allTestClientData[1].clientHandlerPattern)})
+
+	select {
```

Comment:
something like in below example?

for i := 0; i < 2; i++ {
	select {
	case n := <-ch1:
		fmt.Printf("ch1: %d\n", n)

	case n := <-ch2:
		fmt.Printf("ch2: %d\n", n)
	}
}